packages/frontend/src/components/WebsiteRecordList.tsx

```diff
@@ -21,33 +22,43 @@ export function WebsiteRecordList(props: WebsiteRecordListProps) {
   const api = useClient();
 
   useEffect(() => {
-    api
-      ?.GET("/records", {
-        params: {
-          query: {
-            sort: sort ?? "url:asc",
-            filter: filter ?? "",
-            filterBy: filterBy ?? "url",
-            limit: limit ?? 10,
-            offset: offset ?? 0,
-          },
-        },
-      })
-      .then((response) => {
-        setRecords({
-          loading: false,
-          data: response.data ?? {
-            limit: 0,
-            offset: 0,
-            total: 0,
-            records: [],
+    function getRecords() {
```

Comment:
the same refactoring (pulling the function definition to the top of the file) can be done here.